contracts/governance_admin/src/execute/create_pool_creation_proposal.rs

```diff
@@ -0,0 +1,388 @@
+    if let Some(reward_schedules) = &pool_creation_request.reward_schedules {
+        for reward_schedule in reward_schedules {
+            if reward_schedule.start_block_time
+                < env.block.time.plus_seconds(gov_voting_period).seconds()
+            {
+                return Err(ContractError::InvalidRewardScheduleStartBlockTime {});
+            }
+
+            if reward_schedule.end_block_time <= reward_schedule.start_block_time {
+                return Err(ContractError::InvalidRewardScheduleEndBlockTime {});
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Validates if the funds sent by the user are enough to create the pool and other operations
+/// and if yes, then transfers the funds to this contract in case of CW20 tokens since they are not sent along with the message
+/// In case of native tokens, the extra funds are returned back to the user
+pub fn validate_sent_amount_and_transfer_needed_assets(
+    deps: &Deps,
+    env: &Env,
+    sender: &Addr,
+    total_funds_needed: &Vec<Asset>,
+    funds: Vec<Coin>,
+) -> Result<Vec<CosmosMsg>, ContractError> {
+    // return Err(ContractError::Std(StdError::generic_err(funds_str)));
+    let mut messages = vec![];
+
+    // validate that the funds sent are enough for native assets
+    let funds_map = funds
+        .into_iter()
+        .map(|c| (c.denom, c.amount))
+        .collect::<std::collections::HashMap<String, Uint128>>();
+
+    for asset in total_funds_needed {
+        match &asset.info {
+            AssetInfo::NativeToken { denom } => {
+                let amount = funds_map.get(denom).cloned().unwrap_or(Uint128::zero());
+                // TODO: return the extra funds back to the user
+                if amount < asset.amount {
+                    return Err(ContractError::InsufficientFundsSent {
+                        denom: denom.to_string(),
+                        amount_sent: amount,
+                        needed_amount: asset.amount,
+                    });
+                }
+            }
+            AssetInfo::Token { contract_addr } => {
+                // check if the contract has enough allowance to spend the funds
+                let spend_limit = AssetInfo::query_spend_limits(
+                    &contract_addr,
+                    sender,
+                    &deps
+                        .api
+                        .addr_validate(&env.contract.address.to_string())?,
+                    &deps.querier,
+                )?;
+
+                if asset.amount > spend_limit {
+                    return Err(ContractError::InsufficientSpendLimit {
+                        token_addr: contract_addr.to_string(),
+                        current_approval: spend_limit,
+                        needed_approval_for_spend: asset.amount,
+                    });
+                }
+
+                // transfer the funds from the user to this contract
+                let transfer_msg = build_transfer_cw20_from_user_msg(
```

Comment:
CW20 amount in this flow isn't sent along with the message, since that is only possible using the CW20 transfer with a HookMsg. This is fine since if the allowance is sent to a value more than needed, we only transfer the needed amount to us. 

Allowance is set for a single usage only.

Reply:
Done